docs/rules/prefer-array-index-of.md

```diff
@@ -0,0 +1,29 @@
+# Prefer [`Array#indexOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) over [`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) when looking for the index of an item
+
+[`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) is made for more complex criteria. If you are just looking for the index where a given object is present, then the code can be reduced. It concerns any search with a literal, a variable or any expression that hasn't any side effect. However, if the expression you are looking for has side effect, relies on element related to the function (the name of the function, its arguments...), the case is still valid.
+
+This rule is partly fixable.
+
+## Fail
+
+```js
+[].findIndex(x => x === 'foo');
+values.findIndex(x => 'foo' === x);
+values.findIndex(x => {return x === 'foo';});
+['foobar'].findIndex(x => {return 'foo' === x;});
+values.findIndex(function (x) {return x === 'foo';});
+['foobar'].findIndex(function (x) {return 'foo' === x;});
+```
+
+## Pass
+
+```js
+[].findIndex(i => i === list[i]);
+[].findIndex(x => x === 'foo' && isValid);
```

Comment:
This one is not valid cases anymore, please fix it.

Also, please put these in separate markdown code blocks. example https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/error-message.md#fail